erpnext/stock/doctype/delivery_note/delivery_note.js

```diff
@@ -288,3 +288,38 @@ if (sys_defaults.auto_accounting_for_stock) {
 		}
 	}
 }
+
+frappe.ui.form.on('Delivery Note', {
+	set_serial_no: function(doc, cdt, cdn) {
+		var d = frappe.model.get_doc(cdt, cdn);
+		d.qty||(d.qty=1)
+		if(!d.item_code && !d.warehouse && !d.qty) return;
```

Comment:
all the 3 fields are mandatory right? In that case, condition should be based on "OR"